src/mpid/ch4/generic/am/mpidig_am_part_utils.h

```diff
@@ -9,20 +9,35 @@
 #include "ch4_impl.h"
 
 /* Request status used for partitioned comm.
- * Updated as atomic cntr because AM callback and local operation may be
+ *
+ * For sender:
+ *     1. Psend_init:    status = 0, send active message MPIDIG_PART_SEND_INIT
+ *     2. Start (local): status += 1
+ *     3. receive CTS:   status += 1
+ *     4. send data when status == 2 and (all) partitions ready
+ *     5. send complete: status -= 2
+ *     ready for next start.
+ *
+ *     Note that 2 and 3 can take place in any order. Next round step 3 may happen between 4 and 5.
+ *
+ * For receirver:
+ *     1. Precv_init:    status = 0
+ *     2. receive MPIDIG_PART_SEND_INIT: status +=1
+ *     3. match and send CTS
+ *     4. Start (local): status += 1
+ *     5. receive all data and complete: status -= 1
+ *     ready for next start.
+ *
+ *     Note that 2 and 4 can take place in any order, and 3 takes place after 2 or 4, whichever is later.
+ *     After first round, it no longer need matching, so no longer need step 1, 2, and 3.
+ */
```

Comment:
In the second or later round, receiver still needs to send CTS after locally started.